lib/ansible/cli/galaxy.py

```diff
@@ -1451,13 +1458,18 @@ def execute_list_collection(self, artifacts_manager=None):
                     display.vvv("No collections found at {0}".format(collection_path))
                     continue
 
-                # Display header
-                fqcn_width, version_width = _get_collection_widths(collections)
-                _display_header(collection_path, 'Collection', 'Version', fqcn_width, version_width)
+                if output_format in ('yaml', 'json'):
+                    collections_in_paths[collection_path] = [
+                        {'name': c.fqcn, 'version': c.ver} for c in collections
+                    ]
+                else:
+                    # Display header
+                    fqcn_width, version_width = _get_collection_widths(collections)
+                    _display_header(collection_path, 'Collection', 'Version', fqcn_width, version_width)
 
-                # Sort collections by the namespace and name
-                for collection in sorted(collections, key=to_text):
-                    _display_collection(collection, fqcn_width, version_width)
+                    # Sort collections by the namespace and name
+                    for collection in sorted(collections, key=to_text):
+                        _display_collection(collection, fqcn_width, version_width)
```

Comment:
Please dedent this back to keep the nesting sane. It's better to use `continue` to skip it when necessary instead.
```suggestion
                    continue

                # Display header
                fqcn_width, version_width = _get_collection_widths(collections)
                _display_header(collection_path, 'Collection', 'Version', fqcn_width, version_width)

                # Sort collections by the namespace and name
                for collection in sorted(collections, key=to_text):
                    _display_collection(collection, fqcn_width, version_width)
```

Reply:
Done.